File: Leetcode/Algorithms/107. Binary Tree Level Order Traversal II.py
class Solution(object):
    def levelOrderBottom(self, root):
		def search(node, level):
			if(level < len(L)):
				L[level].append(node.val)
			else:
				L.append([node.val])
			if(node.left):
				search(node.left, level + 1) 
			if(node.right):
				search(node.right, level + 1)
			return
		L = []
		if(root):
			search(root, 0)
		return L[::-1]	

File: Algorithms/Tree/107. Binary Tree Level Order Traversal II.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrderBottom(TreeNode* root) {
        vector<vector<int>> res;
        if(!root) return res;
        deque<TreeNode*> queue = {root};
        while(!queue.empty())
        {
            int size = queue.size();
            vector<int> layer;
            for(int i = 0; i < size; i++)
            {
                TreeNode* node = queue.front();
                queue.pop_front();
                layer.emplace_back(node->val);
                if(node->left) queue.emplace_back(node->left);
                if(node->right) queue.emplace_back(node->right);
            }
            res.emplace_back(layer);
        }
        reverse(res.begin(), res.end());
        return res;
    }
};
